util/new_keymap.py

```diff
@@ -40,6 +40,27 @@ def set_logging(debug=None):
     logging.basicConfig(format="%(levelname)s: %(message)s")
 
 
+def cd_qmk_root(expected_files):
```

Comment:
I'm not sure how useful this would be, considering it's somewhat difficult to run this from anywhere other than `qmk_firmware` and `util`.

For simplicity, new_keyboard.sh and my improved new_keymap.sh only check if we're in `util` - if so, `cd` up once - then check for the existence of the `quantum` directory, a reasonably good indicator we're in the right place. They don't check that the current directory is `qmk_firmware` because you can of course git clone the repo into a differently named folder. It probably doesn't need much more than that.

Reply:
@fauxpark : I was considering the scenario (maybe edge case?) where someone is at a keyboard's directory already and wanted to create a new keymap. For that scenario my current script will fail unless they changed to the project's root directory (or to util/), execute the script, then cd back to the created directory. Give so many steps for that possible scenario, I thought maybe I can reduce the steps by making the script smarter with the cd_qmk_root function I wrote.

Also I saw a find_dir function in @skullydazed 's compile_json branch and thought it was a good idea to keep checking upwards until we're in the right directory:
```def find_dir(keyboard):
    """Locate the correct directory for storing a keymap.
    """
    for directory in ['.', '..', '../..', '../../..', '../../../..', '../../../../..']:
        basepath = os.path.normpath(os.path.join('keyboards', keyboard, directory, 'keymaps'))
        if os.path.exists(basepath):
            return basepath```

If you feel those two thoughts above weren't convincing enough for the function, I can definitely revert :)

Reply:
Long term I think we'll have functionality like that in a wrapper script or something. Having code like that all over the place is not my idea of a good time. :)

The `find_dir(keyboard)` function is to locate a keymap dir. To locate the qmk_firmware directory I actually just get the directory for the script and move up one level: https://github.com/qmk/qmk_firmware/compare/compile_json#diff-8cc916123ff32a1a94cfdeb970c4190cR10

Reply:
I like the idea, but I'm just thinking that if you wanted to create a new keymap with this script from a keyboard directory, you'd have to type something like `./../../util/new_keymap.py`, which feels a tad unwieldy to me, and too unlikely to be done by most people to be worth caring about.

If QMK had a bunch more of these little scripts which the install script, say, added to the `PATH`, that would be a different story (and would require even more thorough sanity checking than this 😄).

Reply:
Points taken. Thank you. I'll do a revert later today